Sources/System/FileOperations.swift

```diff
@@ -371,3 +371,109 @@ extension FileDescriptor {
     fatalError("Not implemented")
   }
 }
+
+extension FileDescriptor {
+  /// Opens or creates a file for reading or writing.
+  ///
+  /// - Parameters:
+  ///   - path: The location of the file to open.
+  ///   - at: if `path` is relative, treat it as relative to this file descriptor
```

Comment:
Nit: we should not use prepositions as variable names.

```suggestion
  ///   - base: if `path` is relative, treat it as relative to this file descriptor
```

Or we could just call it `fd`.